libbeat/cmd/instance/metrics.go

```diff
@@ -1,16 +1,39 @@
 package instance
 
 import (
+	"os"
 	"runtime"
 	"time"
 
+	"github.com/elastic/beats/libbeat/common"
 	"github.com/elastic/beats/libbeat/monitoring"
+	sigar "github.com/elastic/gosigar"
+)
+
+type cpuSample struct {
+	time      time.Time
+	procTimes sigar.ProcTime
+}
+
+var (
+	numCores   = runtime.NumCPU()
+	lastSample = cpuSample{
+		time:      time.Now(),
+		procTimes: sigar.ProcTime{},
+	}
 )
 
 func init() {
+	pid := os.Getpid()
+	err := lastSample.procTimes.Get(pid)
```

Comment:
`Get()` is not implemented for all platforms so by adding this you will be limiting the platforms for which Beats can be compiled. This should probably be stubbed in gosigar, but that is currently not the case so you'll need to only use it on the platforms where it's available (linux, windows, openbsd, darwin).